client/index.js

```diff
@@ -16,7 +18,13 @@ class Client {
   onmessage (msg) {
     const data = JSON.parse(msg.data)
 
-    const req = superagent.post(this.target).send(data.body)
+    let target = url.parse(this.target, true)
+    let mergedQuery = Object.assign(target.query, data.query)
```

Comment:
Because these values don't change, they should be `const` instead of `let`. It's a small semantic change, but it does matter at scale through the codebase!

```suggestion
    const mergedQuery = Object.assign(target.query, data.query)
```